### ISSUE TITLE
When I change "Select All", other dropdowns "Select All" gets updated as well

### ISSUE THREAD

**Issue by masteratt** [davidstutz/bootstrap-multiselect#860]
Hello,

I have a init function like so:

`  me.multiselectDropdownUtil = function (elements, customOpts) {
    var before;
    var after;
    var defaultOpts = {
      includeSelectAllOption: true,
      numberDisplayed: 1,
      buttonWidth: "100%",
      allSelectedText: "All",
      onDropdownShown: function (e) {
        before = this.$select.val();
      },
      onDropdownHide: function (e) {
        after = this.$select.val();
        if (!me.arraysEqual(before, after)) {
          // if different i do something. confirmed this is not issue
        }
      }
    };
    var options = $.extend({}, defaultOpts, customOpts);

    elements.multiselect(options);
  };`

I call that like so:

`    me.multiselectDropdownUtil($("#dropdown1"));
    me.multiselectDropdownUtil($("#dropdown2"));`

Front-end:

`                      <select id="dropdown1" multiple="multiple">
                        <option value="0">0</option>
                        <option value="1">1</option>
                      </select>

                      <select id="dropdown2" multiple="multiple">
                        <option value="0">0</option>
                        <option value="1">1</option>
                      </select>
`

Now when I untick the "Select All" checkbox in dropdown1, VISUALLY, the "Select All" checkbox gets unticked in dropdown2 as well. 

Although the DOM still shows for dropdown2:

`<li class="multiselect-item multiselect-all active">`

Note it still has the 'active' class :/

This is super confusing and I'm not sure what is causing it after hours of debugging. I have tried instead of using a method to init them individually.

Any ideas?

**Comment by olaisen81** [comment]
Same problem dude :-(

**Comment by briang123** [comment]
When you change the value of DDL2 does DDL1 change as well?

I think the problem lies in the fact that you're reassigning the function reference with dropdown2 so changes are cascading to other controls.

**Comment by ShinRai1090** [comment]
Faced the same issue, made the below fix (@Ln. _1336_ and _1381_) -

**Before**: `$('li input[value="' + this.options.selectAllValue + '"]')....;` 

**After**: `this.$select.parent().find('input[value="' + this.options.selectAllValue + '"]').....;`

**Comment by briang123** [comment]
Here is what I do that doesn't cause issues across multiple multiselects. 

https://gist.github.com/5d23ff24c2b9b2466f98261c435ea77c

**Comment by briang123** [comment]
Here is the plugin wrapper (cont)

https://gist.github.com/1239268caed3c207b51be8f82250a728

**Comment by davidstutz** [comment]
@masteratt @olaisen81 @ShinRai1090: you need to adapt the `selectAllName` option per multiselect. But I see that this is not documented and should be fixed. Thanks for the discussion.

@briang123: is there a repository for the wrapper? Is it designed to work with jQuery UI?

**Comment by briang123** [comment]
@davidstutz No, this was something I did out of github. I'm going to see about getting this into one.
